symbolic-common/src/types.rs

```diff
@@ -702,10 +730,36 @@ impl<'a> Name<'a> {
         Name {
```

Comment:
Should we make the `new` function take a `mangling` argument instead? 

When constructing a name, we will know in most cases whether the name is mangled or not, particularly:
- Symbol tables are always mangled
- Dwarf can have both, but we know based on which attribute we read
- In PDBs it depends on whether we read it from the GSI or from the MDI.
- In PEs, symbols are always mangled.
- Breakpad symbols are never mangled

This means, we can explicitly invoke the constructor with the mangling and can encourage doing so by not offering a constructor that implicitly fills in unknown. 

If you prefer split constructors, you can also create three for `mangled`, `unmangled` and `unknown`, it seems a bit too much in this case.